toke.c

```diff
@@ -3514,15 +3514,16 @@ S_scan_const(pTHX_ char *start)
                      * of the range is above that value.  We don't count the
                      * end points here, as they will already have been counted
                      * as they were parsed. */
-                    if (range_min >= UTF_CONTINUATION_MARK) {
+                    if (range_min >= UTF_MIN_CONTINUATION_BYTE) {
 
                         /* The whole range is made up of variants */
                         extras = (range_max - 1) - (range_min + 1) + 1;
                     }
-                    else if (range_max >= UTF_CONTINUATION_MARK) {
+                    else if (range_max >= UTF_MIN_CONTINUATION_BYTE) {
 
                         /* Only the higher portion of the range is variants */
-                        extras = (range_max - 1) - UTF_CONTINUATION_MARK + 1;
+                        extras = (range_max - 1)
+                               - UTF_MIN_CONTINUATION_BYTE + 1;
```

Comment:
I don't think this improves much.

From looking at the code range_min and range_max are native code points, either Unicode or EBCIDIC/Unicode (cp1047 in the lower 256 code points rather than Latin-1).

But this compares against what appears to be part of the encoding rather than what looks like a code point.

Perhaps it could use OFFUNI_IS_INVARIANT(), or another macro could define the minimum variant (or the maximum invariant) code point with a name that looks like a code point.

Reply:
done